ui/src/jobs/JobShow.tsx

```diff
@@ -32,6 +36,48 @@ const SuccessField = (props: any) => {
     return (props.record["finished_at"] === null ? <Tooltip title="Running"><JobIcon /></Tooltip> : <BooleanField {...props} />);
 };
 
+const FullButton = ({record}: any) => {
+    const dispatch = useDispatch();
+    const notify = useNotify();
+    const [loading, setLoading] = useState(false);
+    const handleClick = () => {
+        setLoading(true);
+        dispatch(fetchStart()); // start the global loading indicator 
+        fetch(`${apiUrl}/jobs/${record.job_name}/execution/${record.id}`)
+            .then((response) => {
+                if (response.ok) {
+                    notify('Success loading full output');
+                    return response.json()
+                }
+                throw response
+            })
+            .then((data) => {
+                record.output = data.output
+            })
+            .catch((e) => {
+                notify('Error on loading full output', 'warning')
+            })
+            .finally(() => {
+                setLoading(false);
+                dispatch(fetchEnd()); // stop the global loading indicator
+            });
+    };
+    return (
+        <Button 
+            label="Load full output"
+            onClick={handleClick}
+            disabled={loading}
+        >
+            <FullIcon/>
+        </Button>
+    );
+};
+
+const SpecialOutputPanel = ({ id, record, resource }: any) => {
+    // FIXME: hide button if not required
+    return (<div className="execution-output"><div><FullButton record={record} /></div>{record.output || "Empty output"}</div>);
```

Comment:
I'd like to see that if the output was truncated, this is shown to the user, so they know which part is not shown (the tail is shown as opposed to the head).

Reply:
I would like that too as well, as hiding the show button, but as the FIXME says, I couldn’t get around to do it, open to help :)

Reply:
I think it is necessary to have the API return the size of the output when it is truncated, so the UI can determine whether or not everything is displayed. It doesn't have enough information as it stands.
Can't promise I'll have time to help, but I'll see what I can do :)

Reply:
besides the react part to hide or display the button when it should, I'm not even sure where to add said flag in the response.

if I add it to execution struct it will be serialized in the store (with default value) without any real purpose.
any ideas?
I'll give it a try with a `APIExecution` struct (that embeds current Execution)

Reply:
haven't modified the UI yet but updated the matching API endpoint using a new struct `apiEndpoint` which includes `output_truncated` flag.
don't really like the fact that every execution data is copied though..
one other option I can think off is to pass the output size limit itself (not whether it's truncated) and just check output size versus limpt in the UI, doesn't sound ideal though..

Reply:
managed to make use of that variable neatly in the react code, IMO :)